### DISCUSSION TITLE
Security Concerns with Bearer Token Usage for Zitadel API

### DISCUSSION THREAD

**Discussion by harryZhuangTW** [zitadel/zitadel#8764] [Q&A] [answered]
I'm developing a First Party mobile app and have been following the "[Build your own Login UI](https://zitadel.com/docs/guides/integrate/login-ui/username-password)" example. I need to use a Bearer Token to call the ZITADEL API, so I’m generating a JWT private key using a ZITADEL service user and storing it in the iOS Keychain. I obtain the access token using the following request:

```
https://$CUSTOM-DOMAIN/oauth/v2/token
``` 

I then use this access token as the Bearer Token to call the API.

My question is: Is this approach secure? Should I continue using this method, or would it be better to implement a backend server to interact with ZITADEL? Any additional security recommendations or best practices would also be greatly appreciated!

Thank you!

**Comment by fforootd** [comment, marked as answer]
Hey there.

In this case like with a SPA we would recommend that you create a backend for frontend (bff) style service that holds the secrets and not the client.
